lib/modem_key_mgmt/modem_key_mgmt.c

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <at_cmd.h>
+#include <at_cmd_parser/at_cmd_parser.h>
+#include <at_cmd_parser/at_params.h>
+#include <bsd_limits.h>
+#include <modem_key_mgmt.h>
+#include <stdio.h>
+#include <string.h>
+#include <zephyr.h>
+
+#define MODEM_KEY_MGMT_OP_LS "AT%CMNG=1"
+#define MODEM_KEY_MGMT_OP_RD "AT%CMNG=2"
+#define MODEM_KEY_MGMT_OP_WR "AT%CMNG=0"
+#define MODEM_KEY_MGMT_OP_RM "AT%CMNG=3"
+
+#define AT_CMNG_PARAMS_COUNT 5
+#define AT_CMNG_CONTENT_INDEX 4
+
+static char scratch_buf[4096];
+static char at_cmee_strings[2][10] = {"AT+CMEE=0", "AT+CMEE=1"};
+
+static int cmee_active(void)
+{
+	int result;
+	char response[sizeof("+CMEE: X\r\n") + 1];
+
+	memset(response, 0, sizeof(response));
+
+	result = at_cmd_write("AT+CMEE?", response, sizeof(response), NULL);
+	if ((result == 0) && strlen(response) > 0) {
+		if (strchr(&response[6], '1')) {
+			return 1;
+		} else {
+			return 0;
+		}
+	}
+
+	return -EIO;
+}
+
+static int cmee_set(bool enable)
+{
+	return at_cmd_write(at_cmee_strings[(int)enable], NULL, 0, NULL);
+}
+
+static int translate_error(int err)
```

Comment:
What are you really translating here?
This function is passed the return value of `at_cmd_write`; in which case does that function return any of these values?

Reply:
The function returns these errors in case of CMEE enabled. We want to handle these 4.

Reply:
So do I understand correctly that the AT socket will set these values as `errno` ?

Reply:
No, it returns the CME error as response to the AT command issued. The AT socket might fail with an `errno` for other reasons than the AT command itself.